src/Microsoft.FeatureManagement/FeatureManagerSnapshot.cs

```diff
@@ -55,5 +58,60 @@ public Task<bool> IsEnabledAsync<TContext>(string feature, TContext context)
                 feature,
                 (key) => _featureManager.IsEnabledAsync(key, context));
         }
+
+        public Task<bool> IsEnabledAsync(string feature, CancellationToken cancellationToken)
+        {
+            return _flagCache.GetOrAdd(
+                feature,
+                (key) => _featureManager.IsEnabledAsync(key));
+        }
+
+        public Task<bool> IsEnabledAsync<TContext>(string feature, TContext context, CancellationToken cancellationToken)
```

Comment:
I think we better factor in the context as well.

Reply:
if the context is a match, then we can return the same result.

Reply:
Not sure I understand, could you explain what you meant by matching the context?

Reply:
If context isn't equal to the previous context passed, we shouldn't give back the previous result. Looks like it's even an already existing issue. 

I'm thinking maybe we should do something like require TContext to implement IComparable to be able to take advantage of snapshot.

Reply:
It does seem like this is a breaking change though, since it means we have to change all of the interfaces to match the changes in this class.

Reply:
Going to leave this as is for now, was already discussed [here](https://github.com/microsoft/FeatureManagement-Dotnet/issues/87#issuecomment-692172863).